### PR TITLE
add view widget tests

### PR DESCRIPTION
**Pull request by pmario**
This PR adds view widget tests. There are probably some missing, but the date format test is there. 
This PR contains new tests only

### CODE REVIEW

**Comment by BurningTreeC** [comment]
Hi @pmario, are you sure that the date test is correct?

File: editions/test/tiddlers/tests/data/widgets/view/ViewWidget-format-data-template.tid [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
++
+title: Output
+
+<$view tiddler="input" field="modified" format="date"/>
+
+<$view tiddler="input" field="modified" format="date" template="DDth MMM YYYY at hh12:0mmam"/>
++
+title: ExpectedResult
+
+2024 9 20 18:22<p>20th September 2024 at 6:22pm</p>
```
Comment (BurningTreeC) [inline editions/test/tiddlers/tests/data/widgets/view/ViewWidget-format-data-template.tid]: This must be 16:22 at both ends if I'm not wrong

Comment (pmario) [reply editions/test/tiddlers/tests/data/widgets/view/ViewWidget-format-data-template.tid]: That's starnge: All the tests did pass on my end. I'll check again

Comment (BurningTreeC) [reply editions/test/tiddlers/tests/data/widgets/view/ViewWidget-format-data-template.tid]: Did you test on Linux?

Comment (pmario) [reply editions/test/tiddlers/tests/data/widgets/view/ViewWidget-format-data-template.tid]: We may need to specify the time zone

**Comment by pmario** [comment]
The tests are hard to test, because your branch is very much behind upstream. So the testcases do not work in TW. 

Could you merge the current master

**Comment by BurningTreeC** [comment]
> Could you merge the current master

done @pmario

**Comment by pmario** [comment]
@BurningTreeC -- Now all the tests pass on the CI test-sever. But you may need to add some more as you see fit.

**Comment by pmario** [comment]
Close since I can create a PR for upstream now. So the tests can be independent of your view-widget PR

**Pull request closed without merging**